### DISCUSSION TITLE
How do I add an optional navigation property?

### DISCUSSION THREAD

**Discussion by mdufrasne** [OData/AspNetCoreOData#1032] [Q&A]
Let's say I have an entityA

Id,Name,City

And another entityB

Id,Name,Phone

How can I fake a navigation property that connects entityA to entityB

many A for a one or no B

(PS: I know this isn't a proper design, I'm trying to round a circle from an existing schema)

**Comment by habbes** [comment]
Hello @mdufrasne, not sure whether I understand the question. Would adding a navigation property of type B in entity A solve that problem? If you add a nullable single-value navigation property in type A that points to type B, then that's a many-to-one relationship essentially, since multiple A instance can point to the same B instance.

If this would solve the problem, then there are many ways to achieve this. The easiest way is to use the coventional model builder and simply create model classes A and B, and add a property of type B in A. You may also use the non-conventional model builder or the low-level `EdmModel` class to create the navigation properties manually: https://learn.microsoft.com/en-us/odata/webapi/model-builder-abstract
